db/db.go

```diff
@@ -126,3 +153,65 @@ func handleRTDBError(resp *internal.Response) error {
 
 	return err
 }
+
+// parseURLConfig returns the dbURLConfig for the database
+// dbURL may be either:
+//   - a production url (https://foo-bar.firebaseio.com/)
+//   - an emulator URL (localhost:9000/?ns=foo-bar)
+//
+// The following rules will apply for determining the output:
+//   - If the url has no scheme it will be assumed to be an emulator url and be used.
+//   - else If the FIREBASE_DATABASE_EMULATOR_HOST environment variable is set it will be used.
+//   - else the url will be assumed to be a production url and be used.
+func parseURLConfig(dbURL string) (*dbURLConfig, bool, error) {
+	parsedURL, err := url.ParseRequestURI(dbURL)
+	if err == nil && parsedURL.Scheme != "https" {
+		cfg, err := parseEmulatorHost(dbURL, parsedURL)
+		return cfg, true, err
+	}
+
+	environmentEmulatorURL := os.Getenv(emulatorDatabaseEnvVar)
+	if environmentEmulatorURL == "" && err != nil {
+		return nil, false, fmt.Errorf("%s: %w", dbURL, ErrInvalidURL)
+	}
+
+	if environmentEmulatorURL == "" && err == nil {
+		return &dbURLConfig{
+			BaseURL:   dbURL,
+			Namespace: "",
+		}, false, nil
+	}
+
+	parsedURL, err = url.ParseRequestURI(environmentEmulatorURL)
+	if err != nil {
+		return nil, false, fmt.Errorf("%s: %w", environmentEmulatorURL, ErrInvalidURL)
+	}
+	cfg, err := parseEmulatorHost(environmentEmulatorURL, parsedURL)
+	return cfg, true, err
+}
+
+func parseEmulatorHost(rawEmulatorHostURL string, parsedEmulatorHost *url.URL) (*dbURLConfig, error) {
+	if strings.Contains(rawEmulatorHostURL, "//") {
```

Comment:
I saw this behavior in the Python sdk: https://github.com/firebase/firebase-admin-python/blob/37ecf18d0ad09d7402143f22443e957a57ec7a2a/firebase_admin/db.py#L784